### PR TITLE
[Yaml] remove deprecated features

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]
:+1:

File: src/Symfony/Component/Yaml/Inline.php [hunk 1]
```diff
@@ -481,24 +427,20 @@ private static function parseMapping($mapping, $flags, &$i = 0, $references = ar
...
             if (!(Yaml::PARSE_KEYS_AS_STRINGS & $flags)) {
                 $evaluatedKey = self::evaluateScalar($key, $flags, $references);
 
                 if ('' !== $key && $evaluatedKey !== $key && !is_string($evaluatedKey)) {
-                    @trigger_error('Implicit casting of incompatible mapping keys to strings is deprecated since version 3.3 and will throw \Symfony\Component\Yaml\Exception\ParseException in 4.0. Pass the PARSE_KEYS_AS_STRING flag to explicitly enable the type casts.', E_USER_DEPRECATED);
+                    throw new ParseException('Non-string mapping keys are not supported. Pass the Yaml::PARSE_KEYS_AS_STRINGS flag to cast them to strings.', self::$parsedLineNumber + 1, $mapping);
                 }
             }
 
             if (':' !== $key && (!isset($mapping[$i + 1]) || !in_array($mapping[$i + 1], array(' ', ',', '[', ']', '{', '}'), true))) {
```
Comment (GuilhemN) [inline src/Symfony/Component/Yaml/Inline.php]: `':' !== $key` should be removed, right?

File: src/Symfony/Component/Yaml/Inline.php [hunk 2]
```diff
@@ -481,24 +427,20 @@ private static function parseMapping($mapping, $flags, &$i = 0, $references = ar
             // key
             $key = self::parseScalar($mapping, $flags, array(':', ' '), $i, false, array(), true);
```
Comment (GuilhemN) [inline src/Symfony/Component/Yaml/Inline.php]: The last argument should be removed, isn't it?

**Review by GuilhemN — COMMENTED** [review]
:+1:

**Review by GuilhemN — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**